pkgs/development/compilers/ghc/8.10.7.nix

```diff
@@ -231,6 +234,9 @@ stdenv.mkDerivation (rec {
   ] ++ lib.optionals (targetPlatform == hostPlatform && hostPlatform.libc != "glibc" && !targetPlatform.isWindows) [
     "--with-iconv-includes=${libiconv}/include"
     "--with-iconv-libraries=${libiconv}/lib"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "CFLAGS=--target=${stdenv.hostPlatform.config}"
```

Comment:
I think the nixpkgs cc wrapper always passes `-target` and it should have no effect passing it a second time actually:

<https://github.com/NixOS/nixpkgs/blob/2eb8695646429ed33c060c868bf4be5f464257d2/pkgs/build-support/cc-wrapper/default.nix#L301>

AFAIK we don't bypass the cc-wrapper for GHC anywhere.